backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_BOM.java

```diff
@@ -109,4 +120,34 @@ public void validateBOMAttributes(final I_PP_Product_BOM productBom)
 		productPlanningDAO.retrieveProductPlanningForBomVersions(productBOMVersionsId)
 				.forEach(productPlanning -> productBOMService.verifyBOMAssignment(productPlanning, productBom));
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_PP_Product_BOM.COLUMNNAME_ValidFrom, I_PP_Product_BOM.COLUMNNAME_ValidTo })
+	public void preventBOMVersionsOverlapping(final I_PP_Product_BOM productBom)
+	{
+		if (productBom.getValidTo() != null && productBom.getValidTo().before(productBom.getValidFrom()))
+		{
+			throw new AdempiereException(MSG_VALID_TO_BEFORE_VALID_FROM);
+		}
+
+		final List<I_PP_Product_BOM> allBOMVersions = bomDAO.getSiblings(productBom);
```

Comment:
inline this or rename the variable to something like "siblingBOMs"

Reply:
done